mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,310 @@
+    @core.cached_property
+    def keys(self, resolution=28) -> List[dict]:
+        return load_key(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def roman_numerals(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def chords(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords[-1]["time"]
+
+    @core.cached_property
+    def midi_path(self) -> str:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "key": self.keys,
+                "chords": self.chords,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "humdrum_annotated_path": self.humdrum_annotated_path,
+            },
+        )
+
+
+def load_score(path):
+    """Load haydn op20 score with annotations from a file with music21 format (music21.stream.Score).
+
+    Args:
+        path: path to hrm annotations
+
+    Returns:
+        music21.stream.Score: score in music21 format
+
+    """
+    if not os.path.exists(path):
+        raise IOError
+    score = music21.converter.parse(path, format="humdrum")
+    rna = list(score.flat.getElementsByClass("RomanNumeral"))
+    score.remove(rna, recurse=True)
+    return score
+
+
+def load_key(path, resolution=28):
+    """Load haydn op20 key data from a file
+
+    Args:
+        path: path to hrm annotations
+
+    Returns:
+        List[dict]: musical key data and time
+
+    """
+    if not os.path.exists(path):
+        raise IOError
+    score = music21.converter.parse(path, format="humdrum")
+    rna = {rn.offset: rn for rn in list(score.flat.getElementsByClass("RomanNumeral"))}
+    annotations = []
+    for offset, rn in rna.items():
+        if not rn:
+            continue
+        time = int(round(float(offset * resolution)))
+        tonicizedKey = rn.secondaryRomanNumeralKey
+        key = tonicizedKey or rn.key
+        annotations.append({"time": time, "key": key})
+    return annotations
+
+
+def load_midi_path(path):
+    """Load path to midi file of haydn op20 musical piece
+
+    Args:
+        path: path to hrm annotations
+
+    Returns:
+        str: midi file path
+
+    """
+    if not os.path.exists(path):
+        raise IOError
+    midi_path = os.path.splitext(path)[0] + ".midi"
+    if not os.path.exists(midi_path):
+        score = music21.converter.parse(path, format="humdrum")
+        score.write("midi", fp=midi_path)
+    return midi_path
```

Comment:
I'm not sure I don't understand this, can you elaborate a bit what is it doing and why?